lib/src/main/java/com/knuddels/jtokkit/LazyEncodingRegistry.java

```diff
@@ -0,0 +1,26 @@
+package com.knuddels.jtokkit;
+
+import com.knuddels.jtokkit.api.Encoding;
+import com.knuddels.jtokkit.api.EncodingRegistry;
+import com.knuddels.jtokkit.api.EncodingType;
+
+import java.util.Optional;
+
+/**
+ * A lazy initialization implementation of {@link EncodingRegistry}. It does not register any encoding until either the 
+ * {@link #getEncoding(EncodingType)} or {@link #getEncoding(String)} method is called. 
+ * When one of these methods is called, the requested {@link EncodingType} is registered.
+ */
+public class LazyEncodingRegistry extends AbstractEncodingRegistry {
```

Comment:
The getEncodingForModel methods are missing from this registry. This will lead to these methods not being able to make use of the lazy initialization and therefore implicitly relying on `getEncoding(EncodingType)` being called before

This is not the greatest behaviour :smile: I would expect, that those methods work seamless with lazy initialization (eg. also trigger lazy initialization if the requested encoding is not already loaded)

Reply:
Thank you for pointing that out. I will add the `getEncodingForModel` methods back.